static/app/views/projectDetail/projectDetail.tsx

```diff
@@ -48,158 +49,100 @@ type RouteParams = {
-        normalizeUrl({
-          pathname: `/organizations/${organization.slug}/projects/${newlySelectedProject.slug}/`,
-          query: {
-            ...location.query,
-            project: newlySelectedProject.id,
-            environment: undefined,
-          },
-        })
-      );
-    }
-  };
-
-  handleSearch = (query: string) => {
-    const {router, location} = this.props;
-    router.replace({
-      pathname: location.pathname,
-      query: {
-        ...location.query,
-        query,
-      },
-    });
-  };
-
-  tagValueLoader = (key: string, search: string) => {
-    const {location, organization} = this.props;
-    const {project: projectId} = location.query;
-
-    return fetchTagValues({
-      api: this.api,
-      orgSlug: organization.slug,
-      tagKey: key,
-      search,
-      projectIds: projectId ? [projectId] : undefined,
-      endpointParams: location.query,
-    });
-  };
-
-  syncProjectWithSlug() {
-    const {router, location} = this.props;
-    const projectId = this.project?.id;
-
+export default function ProjectDetail({router, location, organization}: Props) {
+  const api = useApi();
+  const params = useParams();
+  const {projects, fetching: loadingProjects} = useProjects();
+  const {selection} = usePageFilters();
+  const project = projects.find(p => p.slug === params.projectId);
+  const {query} = location.query;
+  const hasPerformance = organization.features.includes('performance-view');
+  const hasDiscover = organization.features.includes('discover-basic');
+  const hasTransactions = hasPerformance && project?.firstTransactionEvent;
+  const projectId = project?.id;
+  const isProjectStabilized =
+    defined(project?.id) &&
+    project.id === location.query.project &&
+    project.id === String(selection.projects[0]);
+  const visibleCharts = ['chart1'];
+  const hasSessions = project?.hasSessions ?? null;
+  const hasOnlyBasicChart = !hasPerformance && !hasDiscover && !hasSessions;
+  const title = routeTitleGen(
+    t('Project %s', params.projectId),
+    organization.slug,
+    false
+  );
+
+  const onRetryProjects = useCallback(() => {
+    fetchOrganizationDetails(api, params.orgId, true, false);
+  }, [api, params.orgId]);
+
+  const handleSearch = useCallback(
+    (searchQuery: string) => {
+      router.replace({
+        pathname: location.pathname,
+        query: {
+          ...location.query,
+          query: searchQuery,
+        },
+      });
+    },
+    [router, location.query, location.pathname]
+  );
+
+  const tagValueLoader = useCallback(
+    (key: string, search: string) => {
+      return fetchTagValues({
+        api: api,
+        orgSlug: organization.slug,
+        tagKey: key,
+        search,
+        projectIds: location.query.project ? [location.query.project] : undefined,
+        endpointParams: location.query,
+      });
+    },
+    [api, organization.slug, location.query]
+  );
+
+  const syncProjectWithSlug = useCallback(() => {
```

Comment:
Nit: you could move this into the `useEffect` as it is only used there. Then it would be clearer from the dependency array, when exactly it gets executed.